packages/cli/src/dbt/models.ts

```diff
@@ -312,131 +292,57 @@ export const getModelsFromManifest = (
-    const nodeId = `model.${projectName}.${modelName}`;
-    const node = models.find((model) => model.unique_id === nodeId);
-    if (!node) {
-        throw new ParseError(
-            `Could not find model with name "${modelName}" in project "${projectName}"`,
-        );
-    }
-    let selectedNodes: string[] = [];
-    if (includeParents) {
-        const parents = modelGraph.dependenciesOf(nodeId);
-        selectedNodes = [...selectedNodes, ...parents];
-    }
-    selectedNodes = [...selectedNodes, nodeId];
-    if (includeChildren) {
-        const children = modelGraph.dependantsOf(nodeId);
-        selectedNodes = [...selectedNodes, ...children];
-    }
-    return selectedNodes;
-};
-
-type SelectModelsArgs = {
-    selector: string;
-    projectName: string;
-    models: DbtRawModelNode[];
-    modelGraph: ReturnType<typeof buildModelGraph>;
-};
-const selectModels = ({
-    selector,
-    projectName,
-    models,
-    modelGraph,
-}: SelectModelsArgs) => {
-    const parsedSelector = parseSelector(selector);
-    const { method } = parsedSelector;
-    if (method) {
-        return methodSelector({ method, value: parsedSelector.value, models });
-    }
-    return modelNameSelector({
-        ...parsedSelector,
-        projectName,
-        models,
-        modelGraph,
-    });
-};
-
 type GetCompiledModelsFromManifestArgs = {
-    projectName: string;
-    selectors: string[] | undefined;
     manifest: DbtManifest;
+    select: string[] | undefined;
+    exclude: string[] | undefined;
 };
-export const getCompiledModelsFromManifest = ({
-    projectName,
-    selectors,
+export const getCompiledModelsFromManifest = async ({
     manifest,
-}: GetCompiledModelsFromManifestArgs): CompiledModel[] => {
+    select,
+    exclude,
+}: GetCompiledModelsFromManifestArgs): Promise<CompiledModel[]> => {
     const models = getModelsFromManifest(manifest);
-    const modelGraph = buildModelGraph(models);
-    let nodeIds: string[] = [];
-    if (selectors === undefined) {
-        nodeIds = models.map((model) => model.unique_id);
-    } else {
-        nodeIds = Array.from(
-            new Set(
-                selectors.flatMap((selector) =>
-                    selectModels({ selector, projectName, models, modelGraph }),
-                ),
-            ),
-        );
+    let allModelIds = models.map((model) => model.unique_id);
+
+    if (select !== undefined || exclude !== undefined) {
+        const args: string[] = [];
+
+        if (select && select.length > 0) {
+            args.push('--select', select.join(' '));
+        }
+
+        if (exclude && exclude.length > 0) {
+            args.push('--exclude', exclude.join(' '));
+        }
+
+        try {
+            const { stdout } = await execa('dbt', ['ls', ...args]);
+
+            const filteredModelIds = stdout
+                .split('\n')
+                .filter((line) => line !== '')
+                .map((line) => `model.${line}`);
```

Comment:
will this work if we also use `tags` instead of just model names ?

Reply:
we need to handle the output better, I got all sorts of responses when running `dbt ls` eg: 

![image](https://github.com/lightdash/lightdash/assets/1983672/d783ab49-0049-4d5e-bc52-7d9426837c12)


![image](https://github.com/lightdash/lightdash/assets/1983672/5c7e6da2-a8a4-4b69-bc1d-064ad15b05e6)

![image](https://github.com/lightdash/lightdash/assets/1983672/7bc7cd90-7ea8-4289-8189-0b16120c1a29)

Some dbt versions might use `stderr` instead of `stdout` :fire:

Reply:
@rephus, yes! that works with tags.
I will try the json output which should be way better to deal with